src/Sulu/Component/Contact/Tests/Unit/SmartContent/AccountDataProviderTest.php

```diff
@@ -234,21 +238,21 @@ private function getRepository(
         return $mock->reveal();
     }
 
-    private function createAccount($id, $name, $tags = [])
+    private function createAccount($id, $name, $tags = []): Account
     {
-        $account = $this->prophesize(Account::class);
-        $account->getId()->willReturn($id);
-        $account->getNumber()->willReturn($id);
-        $account->getName()->willReturn($name);
-        $account->getTags()->willReturn($tags);
-        $account->getPlaceOfJurisdiction()->willReturn('');
-        $account->getUid()->willReturn('');
-        $account->getCorporation()->willReturn('');
-        $account->getCreated()->willReturn(new \DateTime());
-        $account->getChanged()->willReturn(new \DateTime());
-        $account->getMedias()->willReturn([]);
-
-        return $account;
+        $entity = new AccountEntity();
+        self::setPrivateProperty($entity, 'id', $id);
+        self::setPrivateProperty($entity, 'number', $id);
+        self::setPrivateProperty($entity, 'name', $name);
+        self::setPrivateProperty($entity, 'tags', new ArrayCollection($tags));
+        self::setPrivateProperty($entity, 'placeOfJurisdiction', '');
+        self::setPrivateProperty($entity, 'uid', '');
+        self::setPrivateProperty($entity, 'corporation', '');
+        self::setPrivateProperty($entity, 'created', new \DateTime());
+        self::setPrivateProperty($entity, 'changed', new \DateTime());
+        self::setPrivateProperty($entity, 'medias', new ArrayCollection([]));
```

Comment:
This is highly in efficient because we are reflecting on this class with every method call. But it's just the tests.

Reply:
@mamazu for properties providing a setter I would directly call the setter method and not go over reflection.

Reply:
Fixed. The only place I have also left with the function is the cache in the `MediaDataProvider`.